devicetypes/smartthings/zwave-siren-tamper.src/zwave-siren-tamper.groovy

```diff
@@ -0,0 +1,283 @@
+			state "default", label: '', action: "alarm.off", icon: "st.secondary.off"
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: '', action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+		valueTile("tamper", "device.tamper", decoration: "flat", width: 2, height: 2) {
+			state "clear", label: 'tamper clear', backgroundColor: "#ffffff"
+			state "detected", label: 'tampered', backgroundColor: "#ff0000"
+		}
+
+		preferences {
+			input name: "alarmLength", type: "number", title: "Alarm length (1-10 min)", range: "1..10"
+			// defaultValue: 10
+			input name: "alarmLEDflash", type: "bool", title: "Alarm LED flash"
+			// defaultValue: false
+			input name: "comfortLED", type: "number", title: "Comfort LED (0-25 x 10 sec.)", range: "0..25"
+			// defaultValue: 0
+			input name: "tamper", type: "bool", title: "Tamper alert"
+			// defaultValue: false
+		}
+
+		main "alarm"
+		details(["alarm", "test", "off", "refresh", "battery", "tamper"])
+	}
+}
+
+def installed() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	sendEvent(name: "tamper", value: "clear", displayed: false)
+
+	response([secure(zwave.basicV1.basicGet()), zwave.batteryV1.batteryGet()])
+}
+
+def updated() {
+	def commands = []
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	if (isYale()) {
+
+		if (!state.alarmLength) state.alarmLength = 10 // default value
+		if (!state.alarmLEDflash) state.alarmLEDflash = 1 // default value
+		if (!state.comfortLED) state.comfortLED = 0 // default value
+		if (!state.tamper) state.tamper = 0 // default value
+
+		log.debug "settings: ${settings.inspect()}, state: ${state.inspect()}"
+
+		Short alarmLength = (settings.alarmLength as Short) ?: 1
+		Boolean alarmLEDflash = (settings.alarmLEDflash as Boolean) ?: 0
+		Short comfortLED = (settings.comfortLED as Short) ?: 0
+		Boolean tamper = (settings.tamper as Boolean) ?: 0
+
+		if (alarmLength != state.alarmLength || alarmLEDflash != state.alarmLEDflash || comfortLED != state.comfortLED || tamper != state.tamper) {
+			state.alarmLength = alarmLength
+			state.alarmLEDflash = alarmLEDflash
+			state.comfortLED = comfortLED
+			state.tamper = tamper
+
+			commands << new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(parameterNumber: 1, size: 1, configurationValue: [alarmLength]).format())
```

Comment:
these don't need to be hubactions, as demonstrated by the basicSet you did below

Also they should be conditionally secured